### PR TITLE
API - Create department model

### PR DESCRIPTION
**Pull request by yonikid15**
# Notes
- Generated a migration that creates department table (id, department_name, and name).
- Generated an eloquent model class.
- Seeded some data copied from [department csv](http://donnees-data.tpsgc-pwgsc.gc.ca/ba1/min-dept/min-dept.csv).
- Added new Department type to the schema and the basic crud query and mutations.

### CODE REVIEW

File: api/database/seeders/DepartmentSeeder.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+class DepartmentSeeder extends Seeder
+{
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+      $departments = [
```
Comment (mnigh) [inline api/database/seeders/DepartmentSeeder.php:17]: Any reason why not all of the data is being seeded from the [department csv](http://donnees-data.tpsgc-pwgsc.gc.ca/ba1/min-dept/min-dept.csv)?

File: api/app/Models/Department.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+
+/**
+ * Class Department
+ *
+ * @property int $id
+ * @property int department_number
```
Comment (mnigh) [inline api/app/Models/Department.php:12]: The [department csv](http://donnees-data.tpsgc-pwgsc.gc.ca/ba1/min-dept/min-dept.csv) uses leadings zeros, for example `"006","Finance (Department of)","Finances (Ministère des)"`. Is the plan to pad these in later for if and when they need to be displayed?

**Review by mnigh — COMMENTED** [review]
Nice work. A few questions to answer before it's ready.
Also, do we need to run `npm run codegen` for the types.d.ts file? If not, should we delete the script from the api folder as well as the types.d.ts file that running the script generates?

File: api/database/seeders/DepartmentSeeder.php (hunk 1 above)
Comment (yonikid15) [reply api/database/seeders/DepartmentSeeder.php:17]: I made another issue for that #486

**Comment by yonikid15** [comment]
> Nice work. A few questions to answer before it's ready.
> Also, do we need to run `npm run codegen` for the types.d.ts file? If not, should we delete the script from the api folder as well as the types.d.ts file that running the script generates?

I'm not sure is it still being referenced anywhere? I just checked and it doesn't seem to be

File: api/app/Models/Department.php (hunk 2 above)
Comment (yonikid15) [reply api/app/Models/Department.php:12]: Ya I believe that's the easiest way (if we plan to display them like that).

**Comment by mnigh** [comment]
> I'm not sure is it still being referenced anywhere? I just checked and it doesn't seem to be

Ok, might be worth creating an issue to remove that script and then file then if we are not using it.

**Review by mnigh — APPROVED** [review]
👍

**Pull request merged**